app/scripts/metamask-controller.js

```diff
@@ -2265,11 +2263,11 @@ export default class MetamaskController extends EventEmitter {
 
     if (useCurrencyRateCheck) {
       this.currencyRateController.stopAllPolling();
+      this.tokenRatesController.stop();
```

Comment:
This line is moved because I believe `useCurrencyRateCheck` is the proper guard.  It was previously gated on the autodetect feature.  But pricing and autodetection are separate settings.